modules/runtime/nuxeo-runtime-migration/src/main/java/org/nuxeo/runtime/migration/MigrationService.java

```diff
@@ -234,4 +237,28 @@ public long getProgressTotal() {
      */
     void runStep(String id, String step);
 
+    /**
+     * @since 11.1
+     */
+    Migration getMigration(String id);
+
+    /**
+     * @since 11.1
+     */
+    List<Migration> getMigrations();
+
+    /**
+     * @throws IllegalStateException if the migration doesn't have exactly one available step to run.
+     * @since 11.1
+     */
+    void run(String migrationId) throws IllegalArgumentException;
```

Comment:
Is it possible to move it next to `runStep`, ideally before it, in order to gather related methods.